docs/src/main/asciidoc/security-authentication-mechanisms.adoc

```diff
@@ -352,13 +352,182 @@ quarkus.http.auth.policy.role-policy-cert.roles-allowed=user,admin     <2>
 
+[[mtls-programmatic-set-up]]
+==== Set up the mutual TLS client authentication programmatically
+
+In the <<mutual-tls>> section we configured the mutual TLS client authentication in the `application.properties` file like this:
+
+----
+quarkus.tls.tls-config-1.key-store.p12.path=server-keystore.p12
+quarkus.tls.tls-config-1.key-store.p12.password=the_key_store_secret
+quarkus.tls.tls-config-1.trust-store.p12.path=server-truststore.p12
+quarkus.tls.tls-config-1.trust-store.p12.password=the_trust_store_secret
+quarkus.http.ssl.client-auth=required   <1>
+quarkus.http.tls-configuration-name=tls-config-1    <2>
+----
+<1> Enable and require the mutual TLS client authentication.
+<2> Use the `tls-config-1` TLS configuration for the HTTP server TLS communication.
+
+The `io.quarkus.vertx.http.security.HttpSecurity` CDI event and the `TlsConfigurationRegistry` bean enable you to configuration the mutual TLS authentication programmatically like in the example below:
+
+.Example TLS configuration
+[source,java]
+----
+package org.acme.http.security;
+
+import io.quarkus.tls.BaseTlsConfiguration;
+import io.vertx.core.net.KeyCertOptions;
+import io.vertx.core.net.KeyStoreOptions;
+import io.vertx.core.net.SSLOptions;
+import io.vertx.core.net.TrustOptions;
+
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+public class MyTlsConfiguration extends BaseTlsConfiguration {
+
+    @Override
+    public KeyCertOptions getKeyStoreOptions() {
+        return new KeyStoreOptions()
+                .setPath("/tmp/certs/server-keystore.p12")
+                .setPassword("the_key_store_secret")
+                .setType("PKCS12");
+    }
+
+    @Override
+    public TrustOptions getTrustStoreOptions() {
+        return new KeyStoreOptions()
+                .setPath("/tmp/certs/server-truststore.jks")
+                .setPassword("the_trust_store_secret")
+                .setType("PKCS12");
+    }
+
+    @Override
+    public SSLOptions getSSLOptions() {
+        SSLOptions options = new SSLOptions();
+        options.setKeyCertOptions(getKeyStoreOptions());
+        options.setTrustOptions(getTrustStoreOptions());
+        options.setSslHandshakeTimeoutUnit(TimeUnit.SECONDS);
+        options.setSslHandshakeTimeout(10);
+        options.setEnabledSecureTransportProtocols(Set.of("TLSv1.3", "TLSv1.2"));
+        return options;
+    }
+
+}
+----
+
+.Example mutual TLS client authentication configuration
+[source,java]
+----
+package org.acme.http.security;
+
+import io.quarkus.vertx.http.security.MTLS;
+import io.quarkus.vertx.http.security.HttpSecurity;
+import io.quarkus.tls.TlsConfigurationRegistry;
+import jakarta.enterprise.event.Observes;
+
+public class MutualTlsClientAuthConfig {
+
+    void configure(@Observes HttpSecurity httpSecurity, TlsConfigurationRegistry tlsConfigRegistry) {
+        httpSecurity.mTLS(MTLS.required("tls-config-1"));    <1>
```

Comment:
I'm not sure if it is possible, but I wonder, can this also work, in addition to what you have:
```
httpSecurity.mTLS("tls-config-1",new MyTlsConfiguration());
```

Otherwise, it looks a little bit disconnected, it is  fine though to get started, but if we could let users do both actions with a single call then it would be nice IMHO.

The API is experimental, so we can do it later